modules/data-entry/src/main/frontend/src/questionnaire/Subject.jsx

```diff
@@ -116,13 +117,29 @@ function Subject (props) {
   const customUrl='/Forms.paginate?fieldname=subject&fieldvalue='
         + encodeURIComponent(data['jcr:uuid']);
 
+  // Recursive function to get a flat list of parents
+  let getParents = (node) => {
+    let output = <React.Fragment>{node.type.label} <Link to={"/content.html" + node["@path"]}>{node.identifier}</Link></React.Fragment>;
+    if (node["parents"]) {
+      let parent = getParents(node["parents"]);
```

Comment:
There can be more than one level of parents.

```suggestion
      let ancestors = getParents(node["parents"]);
```